test/e2e/specs/editor/blocks/cover.spec.js

```diff
@@ -0,0 +1,241 @@
+		page,
+		editor,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		// Locate the Black color swatch.
+		const blackColorSwatch = page.getByRole( 'button', {
+			name: 'Color: Black',
+		} );
+		await expect( blackColorSwatch ).toBeVisible();
+
+		// Get the RGB value of Black.
+		const blackRGB = await blackColorSwatch.evaluate(
+			( node ) => node.style.backgroundColor
+		);
+
+		// Create the block by clicking selected color button.
+		await blackColorSwatch.click();
+
+		// Get the RGB value of the background dim.
+		// The span element implements the colored background.
+		const backgrounDimLocator = page
+			.getByRole( 'document', { name: 'Block: Cover' } )
+			.locator( 'span[aria-hidden="true"]' );
+		const actualRGB = await backgrounDimLocator.evaluate(
+			( node ) => node.style.backgroundColor
+		);
+
+		expect( blackRGB ).toEqual( actualRGB );
+	} );
+
+	test( 'can set background image using image upload on block placeholder', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		const testImage = await getTestImage();
+		const testImageFilename = path.basename( testImage );
+
+		const coverBlock = page.getByRole( 'document', {
+			name: 'Block: Cover',
+		} );
+
+		await coverBlock
+			.getByTestId( 'form-file-upload-input' )
+			.setInputFiles( testImage );
+
+		// Wait for the img's src attribute to be prefixed with http.
+		// Otherwise, the URL for the img src attribute is that of a placeholder.
+		await coverBlock.locator( `img[src^=http]` ).waitFor();
+
+		const backgroundImageURL = await coverBlock
+			.locator( 'img' )
+			.getAttribute( 'src' );
+
+		expect( backgroundImageURL ).toContain( testImageFilename );
+	} );
+
+	test( 'dims background image down by 50% by default', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		const testImage = await getTestImage();
+
+		const coverBlock = page.getByRole( 'document', {
+			name: 'Block: Cover',
+		} );
+
+		await coverBlock
+			.getByTestId( 'form-file-upload-input' )
+			.setInputFiles( testImage );
+
+		const [ backgroundDimColor, backgroundDimOpacity ] =
+			await getBackgroundColorAndOpacity(
+				coverBlock.locator( 'span[aria-hidden="true"]' )
+			);
+		expect( backgroundDimColor ).toBe( 'rgb(0, 0, 0)' );
+		expect( backgroundDimOpacity ).toBe( '0.5' );
+	} );
+
+	test( 'can have the title edited', async ( { page, editor } ) => {
+		const titleText = 'foo';
+
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		// Choose a color swatch to transform the placeholder block into
+		// a functioning block.
+		await page
+			.getByRole( 'button', {
+				name: 'Color: Black',
+			} )
+			.click();
+
+		// Activate the paragraph block inside the Cover block.
+		// The name of the block differs depending on whether text has been entered or not.
+		const coverParagraphLocator = page.getByRole( 'document', {
```

Comment:
If it's inside the cover block, could we use the cover block as the parent locator?